src/lxml/_elementpath.py

```diff
@@ -87,7 +87,10 @@ def xpath_tokenizer(pattern, namespaces=None):
                 except KeyError:
                     raise SyntaxError("prefix %r not found in prefix map" % prefix)
             elif default_namespace and not parsing_attribute:
```

Comment:
I think it's enough to do this:
```suggestion
            elif default_namespace and not parsing_attribute and not tag.isdecimal():
```